### PR TITLE
Core: remove "names" from multidata

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
removes "names", as was planned around 0.4.0

## How was this tested?
minimally, please test too

### CODE REVIEW

File: WebHostLib/tracker.py [hunk 1]
```diff
@@ -264,16 +264,17 @@ def get_static_room_data(room: Room):
     multidata = Context.decompress(room.seed.multidata)
     # in > 100 players this can take a bit of time and is the main reason for the cache
     locations: Dict[int, Dict[int, Tuple[int, int, int]]] = multidata['locations']
-    names: Dict[int, Dict[int, str]] = multidata["names"]
```
Comment (black-sliver) [inline WebHostLib/tracker.py:267]: We might want to push that change in 2 parts. Uploading a new game to current archipelago.gg breaks here :(

File: WebHostLib/tracker.py [hunk 2]
```diff
@@ -264,16 +264,17 @@ def get_static_room_data(room: Room):
...
     custom_items = {}
     if "slot_info" in multidata:
-        games = {slot: slot_info.game for slot, slot_info in multidata["slot_info"].items()}
-        groups = {slot: slot_info.group_members for slot, slot_info in multidata["slot_info"].items()
+        slot_info_dict: Dict[int, NetworkSlot] = multidata["slot_info"]
+        games = {slot: slot_info.game for slot, slot_info in slot_info_dict.items()}
+        groups = {slot: slot_info.group_members for slot, slot_info in slot_info_dict.items()
                   if slot_info.type == SlotType.group}
-
+        names = {0: {slot: slot_info.name for slot, slot_info in slot_info_dict.items()}}
```
Comment (black-sliver) [inline WebHostLib/tracker.py:277]: This is wrong. `names` was List[List[str]] in multidata, but besides awkward typing, this is off by 1. It's being used as `player_name = names[tracked_team][tracked_player - 1]`

**Review by black-sliver — CHANGES_REQUESTED** [review]

File: WebHostLib/tracker.py (hunk 1 above)
Comment (Berserker66) [reply WebHostLib/tracker.py:267]: current main should already be broken in regards to webtracker

**Comment by black-sliver** [comment]
Hm, now it dies in for my test
```
  File "Archipelago/WebHostLib/tracker.py", line 344, in _get_player_tracker
    location_to_area = player_location_to_area[tracked_player]
```
`multidata["checks_in_area"]` probably not fine? Is there already a PR open to fix that and I just missed it?

**Comment by Berserker66** [comment]
I'll try to remember to actually test the tracker tomorrow. I only tested the MultiServer and was apparently falsely hoping the tracker would mostly be prepared for the change.

**Review by ThePhar — APPROVED** [review]

**Comment by ThePhar** [comment]
Confirmed trackers working again.

**Review by black-sliver — APPROVED** [review]
The stuff that didn't work for me earlier seems to work now :+1:

**Pull request merged**